core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java

```diff
@@ -354,6 +323,39 @@ private void cancelTimeoutOrLog(@Nullable Throwable cause,
             logger.warn(logMsg.append(':').toString(), cause);
         }
 
+        void initTimeout() {
+            initTimeout(responseTimeoutMillis);
+        }
+
+        private TimeoutTask newTimeoutTask() {
+            return new TimeoutTask() {
+                @Override
+                public boolean isReady() {
+                    return delegate.isOpen();
+                }
+
+                @Override
+                public boolean canReschedule() {
+                    return state != State.DONE;
```

Comment:
Question: Can't we combine `isReady()` and `canReschedule()`?

Reply:
This interface was designed from the previous code. I think the combined API can cover current case. I wonder that there is any chance a timeout should be scheduled once?

Reply:
If we can combine them how about renaming to `canSchedule`?

Reply:
OK, let's simplify it.